packages/integrations/tailwind/src/index.ts

```diff
@@ -1,101 +1,50 @@
-		} catch {
-			postcssConfigResult = null;
-		}
-	}
-	return postcssConfigResult;
-}
-
-async function getViteConfiguration(
-	tailwindConfigPath: string | undefined,
-	nesting: boolean,
-	root: string,
-	postcssInlineOptions: CSSOptions['postcss'],
-): Promise<Partial<UserConfig>> {
-	// We need to manually load postcss config files because when inlining the tailwind and autoprefixer plugins,
-	// that causes vite to ignore postcss config files
-	const postcssConfigResult = await getPostCssConfig(root, postcssInlineOptions);
-
-	const postcssOptions = postcssConfigResult?.options ?? {};
-	const postcssPlugins = postcssConfigResult?.plugins?.slice() ?? [];
-
-	if (nesting) {
-		const tailwindcssNestingPlugin = (await import('tailwindcss/nesting/index.js')).default;
-		postcssPlugins.push(tailwindcssNestingPlugin());
-	}
-
-	postcssPlugins.push(tailwindPlugin(tailwindConfigPath));
-	postcssPlugins.push(autoprefixerPlugin());
-
-	return {
-		css: {
-			postcss: {
-				...postcssOptions,
-				plugins: postcssPlugins,
-			},
-		},
-	};
-}
+import tailwindcss from '@tailwindcss/vite';
+import { fileURLToPath } from 'node:url';
+import { writeFileSync } from 'node:fs';
 
 type TailwindOptions = {
 	/**
 	 * Path to your tailwind config file
-	 * @default 'tailwind.config.mjs'
+	 * @deprecated It is recommended to use the `@config` directive, see https://tailwindcss.com/docs/upgrade-guide#using-a-javascript-config-file
+	 * @default './tailwind.config.mjs'
 	 */
 	configFile?: string;
 	/**
 	 * Apply Tailwind's base styles
-	 * Disabling this is useful when further customization of Tailwind styles
-	 * and directives is required. See {@link https://tailwindcss.com/docs/functions-and-directives#tailwind Tailwind's docs}
-	 * for more details on directives and customization.
+	 * @deprecated It is recommended to use a css file to configure Tailwind, see https://tailwindcss.com/docs/upgrade-guide#removed-tailwind-directives
 	 * @default true
 	 */
 	applyBaseStyles?: boolean;
-	/**
-	 * Add CSS nesting support using `tailwindcss/nesting`. See {@link https://tailwindcss.com/docs/using-with-preprocessors#nesting Tailwind's docs}
-	 * for how this works with `postcss-nesting` and `postcss-nested`.
-	 */
-	nesting?: boolean;
 };
 
-export default function tailwindIntegration(options?: TailwindOptions): AstroIntegration {
-	const applyBaseStyles = options?.applyBaseStyles ?? true;
-	const customConfigPath = options?.configFile;
-	const nesting = options?.nesting ?? false;
+export default function tailwindIntegration({
+	applyBaseStyles = true,
+	configFile,
+}: TailwindOptions = {}): AstroIntegration {
+	if (applyBaseStyles && !configFile) {
+		configFile = './tailwind.config.mjs';
+	}
 
 	return {
 		name: '@astrojs/tailwind',
 		hooks: {
-			'astro:config:setup': async ({ config, updateConfig, injectScript }) => {
-				// Inject the Tailwind postcss plugin
+			'astro:config:setup': async ({ config, updateConfig, injectScript, createCodegenDir }) => {
```

Comment:
Let's log a deprecation warning somewhere, so users know that this integration is deprecated

Reply:
I think @matthewp did not want a warning yet? But I agree with you